runtime/lua/vim/lsp/buf.lua

```diff
@@ -250,7 +250,7 @@ end
 ---name using |input()|.
 function M.rename(new_name)
   local params = util.make_position_params()
-  local prepare_rename = function(err, _, result)
+  function prepare_rename(err, _, result)
```

Comment:
Should this not be `local function prepare_rename`? Or is the function local by default by virtue of it being defined inside a function scope?

Reply:
`local function` is a good practice--my mistake. https://stackoverflow.com/a/51052827/152142
```suggestion
  local function prepare_rename(err, _, result)
```